src/api-v1.js

```diff
@@ -299,6 +317,20 @@ api.declare({
 
   let modDate = new Date();
 
+  // We want to make sure that all AMIs that we are submitting are valid
+  let missing = [];
+  await Promise.all(input.regions.map(async (def) => {
+    if (await amiExists(this.ec2, def.launchSpec.ImageId, def.region)) {
+      missing.push({imageId: def.launchSpec.ImageId, region: def.region});
+    }
+  }));
+  if (missing.length > 0) {
+    return res.status(400).json({
```

Comment:
nit, if you are on a recent tc-lib-api, there is a reportError function which will do great here...